packages/react-components/react-dialog/src/components/DialogBody/useDialogBody.ts

```diff
@@ -16,9 +16,12 @@ export const useDialogBody_unstable = (props: DialogBodyProps, ref: React.Ref<HT
     components: {
       root: 'div',
     },
-    root: getNativeElementProps(props.as ?? 'div', {
-      ref,
-      ...props,
-    }),
+    root: resolveShorthand(
+      getNativeElementProps(props.as ?? 'div', {
```

Comment:
Do we need to apply this for `root` slot? We don't have documented support for render functions on root slots.

Reply:
There might be valid use cases for a render function on the `root` slot. One case that @khmakoto mentioned to me was that a user might want to use a root slot render function on `<Link>` to use a router Link as the root component (instead of `<a>`).

Reply:
If we would like to support that we need to update types, fix all other usages and document it. Currently, it throws a TS error with latest types and works only due "`any`" in `ReactNode`: https://codesandbox.io/s/cocky-architecture-osickm

Reply:
Types work fine actually
![image](https://user-images.githubusercontent.com/5483269/231989645-d8441760-761c-4447-86b2-e6134db26db9.png)


What you're seeing there in the codesandbox is related to this https://github.com/microsoft/fluentui/issues/27090, this still needs to be investigated but it is not type related.

Reply:
Well, I guess I'll keep that to another PR?! 😅

But let's keep in mind that treating root as something else than just a slot will cause some problems https://github.com/microsoft/fluentui/pull/27475#discussion_r1166512670

Reply:
@bsunderhus Indeed, let's handle it separately. Can you please create an issue for an issue to create an RFC related to this topic?